saas/models/saas_operator.py

```diff
@@ -52,3 +54,33 @@ def get_db_name(self, db):
         # TODO: use mako for url templating
         self.ensure_one()
         return self.db_url_template.format(db_id=db.id)
+
+    def get_auth_urls(self, db):
+        self.ensure_one()
+        if self.type == 'local':
+            master_url = self.env['ir.config_parameter'].get_param('web.base.url')
+            build_url = self.get_db_url(db)
+        return master_url, build_url
+
+    def build_execute_kw(self, build, code):
+        if self.type == 'local':
+            db = sql_db.db_connect(build.name)
+            registry(build.name).check_signaling()
+            with api.Environment.manage(), db.cursor() as cr:
+                env = api.Environment(cr, SUPERUSER_ID, {})
+                action = env['ir.actions.server'].create({
+                    'name': 'Local Code Eval',
+                    'state': 'code',
+                    'model_id': 1,
+                    'code': code,
+                })
+                action.run()
```

Comment:
This code still depends on operator type. If we keep it this way you need to copy past "action create" and "action run" for every operator.